### DISCUSSION TITLE
purge block retraction, dual extruder

### DISCUSSION THREAD

**Discussion by phongshader** [supermerill/SuperSlicer#2730] [Q&A] [answered]
I'm switching over to Superslicer and I'm having a hard time getting my head wrapped around getting multi extruder single nozzle settings. What is happening, for me with no filament swap, is the current filament retracts to parking position when transitioning from part to purge tower. This seems to be a huge waste of time and also causes filament jams in my case. Is this by design? Wouldn't a simple retraction suffice in this situation? Under Printer Settings > Single Extruder MM Settings there are several settings, Cooling tube position, cooling tube length, and filament parking position. Filament parking position is self evident...where the filament retracts to when a material change happens and parks when another filament is being used.  I'm not sure what the other 2 are for. I have my cooling tube position set to 35mm (aproximately the middle of the hotend heatsink) and parking position is 95mm. Is there a way to get the settings so that when there is no filament change a simple retraction is used instead of retracting to the park position when transitioning to the purge block?

**Comment by schnello** [comment]
Could you upload the gcode?
In my case: Multi extruder and single nozzle it work as expected. Park filament only if there is a real swap

**Comment by phongshader** [comment]
Here is an example
[mmcubecalibration.gcode.zip](https://github.com/supermerill/SuperSlicer/files/8558495/mmcubecalibration.gcode.zip)
I opened it up in vscode and did a search and replace of E-95 and E95 and replaced with E-5 and E5 and the print completed much faster.

**Reply by phongshader** [reply]
printer settings
<img width="646" alt="Screen Shot 2022-04-25 at 5 56 23 PM" src="https://user-images.githubusercontent.com/38082624/165197917-0c9373c9-4f45-4031-b1f4-f68a14f8e183.png">

**Comment by supermerill** [comment]
If you don't have an automatic mmu with only one extruder (like the ones made by prusa), don't activate "single extruder multi material'.

**Comment by schnello** [comment]
Please set this value to 0
![image](https://user-images.githubusercontent.com/6481341/165764236-85e6a248-0e09-4d5e-8bef-a03362d4a663.png)

**Reply by phongshader** [reply]
I will try that...

**Reply by phongshader** [reply]
Setting minimum retraction to 0 works as long as there is no filament change however on filament change there has to be a retraction of filament1 to allow the loading of filament2, 95mm. This is not a solution.

**Reply by phongshader** [reply]
maybe tis is better handled with tool change gcode

**Reply by phongshader** [reply]
OK, lightbulb goes off in head...use Single Extruder Multi Material but set Minimum retraction to 0. So the way to set up dual extruder single nozzle is to **_SELECT_** Single Extruder Multi Material in Printer Settings > General, then Printer Settings > Single Extruder MM Setup set the correct values for Cooling tube position, cooling tube length, and filament parking, then Printer Settings > Extruder1 and Extruder2 set minimum retraction to 0. This is working for me. THANK YOU for all your help! This should be added to some documentation somewhere.

**Comment by phongshader** [comment, marked as answer]
So the way to set up dual extruder single nozzle is to SELECT Single Extruder Multi Material in Printer Settings > General, then Printer Settings > Single Extruder MM Setup set the correct values for Cooling tube position, cooling tube length, and filament parking, then Printer Settings > Extruder1 and Extruder2 set minimum retraction to 0.
